modules/nextflow/src/main/groovy/nextflow/processor/TaskProcessor.groovy

```diff
@@ -972,8 +975,15 @@ class TaskProcessor {
             if( error instanceof Error ) throw error
 
             // -- retry without increasing the error counts
-            if( task && (error.cause instanceof NodeTerminationException || error.cause instanceof CloudSpotTerminationException) ) {
-                if( error.cause instanceof NodeTerminationException )
+            if( task && (error.cause instanceof NodeTerminationException
+                    || error.cause instanceof CloudSpotTerminationException
+                    || error instanceof K8sOutOfCPUException
+                    || error instanceof K8sOutOfMemoryException
+            ) ) {
+                if( error.cause instanceof NodeTerminationException
+                        || error instanceof K8sOutOfCPUException
+                        || error instanceof K8sOutOfMemoryException
+                )
```

Comment:
To keep these condition checks from getting out of hand, why don't we make NodeTerminationException and your two new exceptions inherit from a common class. Not sure what the name should be but something that denotes "errors that should trigger a retry instead of workflow termination".

Reply:
I introduced an interface "TemporaryProblem", good point!